dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -22,6 +22,36 @@ def clever
     end
   end
 
+  # POST /users/auth/maker_google_oauth2
+  def maker_google_oauth2
+    secret = Encryption.decrypt_string_utf8(params[:secret_code])
+    time = DateTime.strptime(secret.slice!(0..19), '%Y%m%dT%H%M%S%z')
+    time_difference = (Time.now - time) / 1.minute
+
+    # Reject - code was generated more than 5 minutes ago or incorrect provider
+    if time_difference >= 5
+      flash.now[:alert] = "Token has expired. Please log in again for a new token."
+      render 'maker/login_code'
+      return
```

Comment:
super nit / shortcut:

```suggestion
      return render 'maker/login_code'
```